wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java

```diff
@@ -106,6 +106,26 @@
    */
   public static final String PERSIST_VOLUMES_ATTRIBUTE = "persistVolumes";
 
+  /**
+   * This attribute configures a workspace running on OpenShift to attempt to work around the
+   * OpenShift default of starting containers with an arbitrary UID. Should be set/read from {@link
+   * WorkspaceConfig#getAttributes()}.
+   *
+   * <p>Value is expected to be boolean; if it is set to 'true', then the recipe containers in the
```

Comment:
Can you elaborate on why only recipe containers should be affected?

Reply:
There are a few reasons I want to keep it to recipe containers only.

1. It's kind of a rough proxy for only affecting user-specified containers; we don't want to run the provisioner for broker containers.

1. I tried to keep this PR a relatively minimal set of changes. The issue we're trying to solve affects workspace containers primarily and so I focussed on those. If similar issues exist for typical sidecar containers then we could include those as well. I'd generally prefer to not rewrite commands unless we have to since it introduces potential errors and issues
   - Additionally, our plugin runner containers don't specify a command/argument so this wouldn't work (it would actually break workspaces since the entrypoint would no longer be terminating)

1. Since plugin runner containers and similar are 'write once, run everywhere' containers, I think a better solution if a similar change is needed is to create the containers with the fixes built in. We control those dockerfiles and entrypoints already so a kind of sloppy work around is not necessary.

This does bring up a good point though; I think this PR as it currently is will break the devfile demo, since the `db` container doesn't list an entrypoint. I'll have to address that (only do the overwriting if there is an command specified.